app/routes/recipes-generator.jsx

```diff
@@ -44,54 +36,209 @@ export default function RecipiesGenerator() {
+              const recipeNameResponse = await api.getRecipeNameFromId(
+                recipe.recipe_id
+              );
+              return {
+                ...recipe,
+                title: recipeNameResponse.success
+                  ? recipeNameResponse.title
+                  : "Unknown Recipe",
+              };
+            })
+          )
+        : [];
+      setLastConsumedRecipes(recipesWithTitles);
+    } catch (error) {
+      console.error("Error fetching Last Consumed Recipes:", error.message);
+      setLastConsumedRecipes([]);
+    }
+  };
+
+  const fetchFavoriteRecipes = async () => {
+    try {
+      const response = await api.getFavoriteRecipes();
+      const recipesWithTitles = response.length
+        ? await Promise.all(
+            response.map(async (recipe) => {
+              const recipeNameResponse = await api.getRecipeNameFromId(
+                recipe.recipe_id
+              );
+              return {
+                ...recipe,
+                title: recipeNameResponse.success
+                  ? recipeNameResponse.title
+                  : "Unknown Recipe",
+              };
+            })
+          )
+        : [];
+      setFavoriteRecipes(recipesWithTitles);
+    } catch (error) {
+      console.error("Error fetching Favorite Recipes:", error.message);
+      setFavoriteRecipes([]);
+    }
+  };
+
+  useEffect(() => {
+    if (!isAuthenticated) {
+      loginWithRedirect();
+      return;
+    }
+    
+    if (tokenReady) {
+      fetchLastConsumedRecipes();
+      fetchFavoriteRecipes();
+    }
+  }, [tokenReady]);
+
+  const toggleFavorite = async (recipeId) => {
+    const isFavorite = favoriteRecipes.some(
+      (recipe) => recipe.recipe_id === recipeId
+    );
+
+    if (isFavorite) {
+      setFavoriteRecipes((prev) =>
+        prev.filter((recipe) => recipe.recipe_id !== recipeId)
+      );
+      try {
+        await api.removeRecipeFromFavorites(recipeId);
+      } catch (error) {
+        console.error(
+          `Error removing recipe ${recipeId} from favorites:`,
+          error.message
+        );
+      }
+    } else {
+      const recipeToAdd =
+        lastConsumedRecipes.find((recipe) => recipe.recipe_id === recipeId) || {
+          recipe_id: recipeId,
+        };
+      setFavoriteRecipes((prev) => [...prev, recipeToAdd]);
+      try {
+        await api.addRecipeToFavorites(recipeId);
+      } catch (error) {
+        console.error(
+          `Error adding recipe ${recipeId} to favorites:`,
+          error.message
+        );
+      }
+    }
+  };
+
   return (
     <div className="generalContainer flex">
       <SideBar />
       <div className="ContainerBody relative h-[1100px] grow bg-[#E5E9F0] p-[60px] z-[0] overflow-y-hidden">
-        <h1 className="text-3xl text-[#182F40] font-bold">
-          {headerText}<span className="text-7xl font-extralight">{headerSpanText}</span>
+        <h1 className="text-3xl font-bold text-[#182F40] font-kantumruy">
```

Comment:
I think that this breaks the formatting requirements for mantaining a uniform font throught the project.
If the font-change is **necessary** and **part of the figma**, then ignore!